test/net/sourceforge/kolmafia/objectpool/ConcoctionTest.java

```diff
@@ -30,61 +52,51 @@ public void itShouldSortUsables() {
+        msg = "comparing (quasi symmetry)" + idsAndMaps.get(i) + "and " + idsAndMaps.get(j);
+        assertEquals(sgn(result[i][j]), -sgn(result[j][i]), msg);
       }
     }
   }
 
-  // x.compareTo(y)==0 implies
-  //	  that sgn(x.compareTo(z)) == sgn(y.compareTo(z)), for all z.
+  // tests the portion of the contract that says (x.compareTo(y)==0) == (x.equals(y))
   @Test
-  @Disabled("Nested for-loops are slow...")
-  public void itShouldBePreserveEquality() {
-    LockableListModel<Concoction> first = ConcoctionDatabase.getUsables();
-    LockableListModel<Concoction> second = ConcoctionDatabase.getUsables();
-    LockableListModel<Concoction> third = ConcoctionDatabase.getUsables();
-    for (Concoction acon : first) {
-      for (Concoction bcon : second) {
-        if (acon.compareTo(bcon) == 0) {
-          for (Concoction ccon : third) {
-            String msg = acon.toString() + " * " + bcon.toString() + " * " + ccon.toString();
-            int x = sgn(acon.compareTo(ccon));
-            int y = sgn(bcon.compareTo(ccon));
-            assertEquals(sgn(x), sgn(y), msg);
-          }
+  public void compareToShouldBeEqualForEquals() {
+    for (int i = 0; i < maxIndex; i++) {
+      msg = "comparing (equality)" + idsAndMaps.get(i) + "and " + idsAndMaps.get(i);
+      assertEquals(0, result[i][i], msg);
+      for (int j = 0; j < maxIndex; j++) {
+        if (result[i][j] == 0) {
+          msg = "comparing (equality)" + idsAndMaps.get(i) + "and " + idsAndMaps.get(j);
+          assertEquals(idsAndMaps.get(i), idsAndMaps.get(j), msg);
         }
       }
     }
   }
 
-  // (x.compareTo(y)>0 && y.compareTo(z)>0) implies x.compareTo(z)>0.
+  // x.compareTo(y)==0 implies
+  //	  that sgn(x.compareTo(z)) == sgn(y.compareTo(z)), for all z.
   @Test
-  @Disabled("Test takes too much resources.  Needs optimization")
-  public void isBT() {
-    LockableListModel<Concoction> first = ConcoctionDatabase.getUsables();
-    first.sort();
-    Concoction[] cons = first.toArray(new Concoction[0]);
-    for (Concoction acon : cons) {
-      for (Concoction bcon : cons) {
-        int x = sgn(acon.compareTo(bcon));
-        if (x > 0) {
-          for (Concoction ccon : cons) {
-            String msg = acon.toString() + " * " + bcon.toString() + " *" + ccon.toString();
-            int y = sgn(bcon.compareTo(ccon));
-            if (y > 0) assertTrue(sgn(acon.compareTo(ccon)) > 0, msg);
+  public void itShouldBeTransitive() {
+    for (int i = 0; i < maxIndex; i++) {
+      // Don't have to check whole matrix
+      for (int j = i; j < maxIndex; j++) {
+        if (result[i][j] > 0) {
+          for (int k = 1; k < maxIndex; k++) {
+            if (result[j][k] > 0) {
+              msg =
+                  "comparing (transitive)"
+                      + idsAndMaps.get(i)
+                      + "and "
+                      + idsAndMaps.get(j)
+                      + "and "
+                      + idsAndMaps.get(k);
```

Comment:
You're missing several spaces.

This will print as: "comparing (transitive)seal-clubbing cluband seal toothand bar skin"

You're also not using msg anywhere in this function.

Reply:
Fixed